rgbd_depth_correction/src/depth_calibration.cpp

```diff
@@ -0,0 +1,505 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2015, Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <depth_calibration/depth_calibration.h>
+#include <target_finder/target_locater.h>
+#include <boost/thread/locks.hpp>
+#include <sstream>
+
+#include <tf/tf.h>
+#include <geometry_msgs/Transform.h>
+
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl/point_cloud.h>
+#include <pcl/io/pcd_io.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/calib3d/calib3d.hpp>
+#include <opencv2/highgui/highgui.hpp>
+
+
+
+const unsigned int DepthCalibrator::VERSION_NUMBER_ = 1;
+
+DepthCalibrator::DepthCalibrator(ros::NodeHandle& nh)
+{
+  nh_ = nh;
+  ros::NodeHandle pnh("~");
+
+  // get parameters for services, topics, and filename
+  pnh.param<std::string>("filename", filename_, "");
```

Comment:
If parameters are required (no defaults), then some error checking and reporting is necessary